src/platform_impl/linux/wayland/header.rs

```diff
@@ -0,0 +1,44 @@
+use gtk::{
+  glib::{self},
+  prelude::*,
+  ApplicationWindow, HeaderBar,
+};
+
+pub struct WlHeader;
+
+impl WlHeader {
+  pub fn setup(window: &ApplicationWindow) {
+    let header = HeaderBar::builder()
+      .show_close_button(true)
+      .decoration_layout("menu:minimize,maximize,close")
+      .build();
+
+    window.set_titlebar(Some(&header));
+    Self::connect_move_window(&header, &window);
+  }
+
+  fn connect_move_window(header: &HeaderBar, window: &ApplicationWindow) {
+    let header_weak = header.downgrade();
+    window.connect_button_press_event(move |window, event| {
```

Comment:
why not just `header.connect_button_press_event`?

Reply:
and then you probably don't need to check 
```
 if x >= window_x as f64
            && x <= (window_x + window_width) as f64
            && y >= window_y as f64
            && y <= (window_y + header_height) as f64
```

Reply:
I tried using `header.connect_button_press_event`. Unfortunately, it didn’t seem to recognize the full width of the window.
I will try to find a way to get the width of the header.

![header-drag](https://github.com/user-attachments/assets/56e8b9e2-7647-4325-b9de-0280be31ec95)

Reply:
maybe try checking the width of the header bar using GTK inspector https://developer.gnome.org/documentation/tools/inspector.html

Reply:
Wow, that’s really useful. I wrapped the header in an event box, and the drag event works perfectly.
![event-box-work](https://github.com/user-attachments/assets/863a4209-1313-4cf4-9ad8-55d96c4bed7e)

Reply:
It seems that using `set_titlebar()` passes events through to the `gtkWindow`.
See: https://stackoverflow.com/questions/44670188/gtk3-widget-mouse-events-pass-through

This is likely the reason why `header.connect_button_press_event` is not working. 
Ultimately, we need to decide whether to use `window.connect_button_press_event` or stick with `event_box`.

Both options work the same.

Reply:
EventBox seems okay